docs/design-documents/tools/cmake.md

```diff
@@ -1,89 +1,85 @@
 # CMake Mbed OS
 
+Requirements:
+- CMake 3.13 and higher
+- mbed-tools (python 3.6 and higher)
+
 Two steps approach:
 
 - Mbed OS core CMake
 - Application CMake
 
-All what can be defined should be in CMake files and configuration (app/mbed .json files). Our build system would parse the configuration, create rules and generate top level CMake file that would include others (application CMake files plus also Mbed OS core CMake).
+Everything that can be defined should be in CMake files and configuration (mbed_app/mbed_lib.json) files. Our build system would parse the configuration and generate a CMake config file.
 
 To stay backward compatible, we create common rules as follows.
 
-We replace current `TOOLCHAIN_`, `FEATURE_`, `TARGET_` by providing `add_subdirectory` macros. There would be `if_target()`, `if_feature()`, `_if_toolchain()` macros for better readability.
-
-Example, to add stm_directory only for STM targets (current rule TARGET_STM):
-
-```
-add_subdirectory_if_target(STM stm_directory)
-
-```
+We use target labels to include subfolders. A custom CMake function goes through the target labels. If there is a folder in the current CMake source directory matching a target label, it is included. 
```

Comment:
You don't need the comma before "and features".

Reply:
Oxford Comma: 
https://www.grammarly.com/blog/what-is-the-oxford-comma-and-why-do-people-care-so-much-about-it/

Reply:
Yes, I know what it is. I was always taught to avoid them :). It's used more in U.S English than British English, where it isn't considered a standard.

Reply:
Tomato tomato...

Reply:
Let's go with this change, then? I think it's worth adding, even with the fancy extra comma...